dataactvalidator/config/sqlRules.csv

```diff
@@ -0,0 +1,4 @@
+rule_sql,rule_label,rule_description,rule_error_message,rule_cross_file_flag,file_type,severity_name
+"SELECT af.row, af.bydirectreimbursablefundingsource FROM award_financial AS af WHERE job_id = {} AND (ussgl480100_undeliveredordersobligationsunpaid_fyb IS NOT NULL OR ussgl480100_undeliveredordersobligationsunpaid_cpe IS NOT NULL OR ussgl483100_undeliveredordersobligtransferredunpaid_cpe IS NOT NULL OR ussgl488100_upadjsprioryearundelivordersobligunpaid_cpe IS NOT NULL OR ussgl490100_deliveredordersobligationsunpaid_fyb IS NOT NULL OR ussgl490100_deliveredordersobligationsunpaid_cpe IS NOT NULL OR ussgl493100_deliveredordersobligstransferredunpaid_cpe IS NOT NULL OR ussgl498100_upadjsprioryeardeliveredordersobligunpaid_cpe IS NOT NULL OR ussgl480200_undeliveredordersobligationsprepaidadvanced_fyb IS NOT NULL OR ussgl480200_undeliveredordersobligationsprepaidadv_cpe IS NOT NULL OR ussgl483200_undeliveredordersobligtransferredppdadv_cpe IS NOT NULL OR ussgl488200_upadjsprioryrundelivordersobligprepaidadv_cpe IS NOT NULL OR ussgl490200_deliveredordersobligationspaid_cpe IS NOT NULL OR ussgl490800_authorityoutlayednotyetdisbursed_fyb IS NOT NULL OR ussgl490800_authorityoutlayednotyetdisbursed_cpe IS NOT NULL OR ussgl498200_upadjsprioryrdelivordersobligpaid_cpe IS NOT NULL) AND COALESCE(bydirectreimbursablefundingsource,'') NOT IN ('R', 'D')",B12,"Reimbursable flag indicator is required when reporting obligation or outlay USSGL account balances (excluding downward adjustments SGL accounts). Valid values are ""R"" for ""ReimbursableFundingSource"" and ""D"" for ""DirectFundingSource"".","Reimbursable flag indicator is required when reporting obligation or outlay USSGL account balances (excluding downward adjustments SGL accounts). Valid values are ""R"" for ""ReimbursableFundingSource"" and ""D"" for ""DirectFundingSource"".",,award_financial,fatal
+"SELECT approp.row, approp.tas, approp.grossoutlayamountbytas_cpe, SUM(op.grossoutlayamountbyprogramobjectclass_cpe) as grossoutlayamountbyprogramobjectclass_cpe_total FROM appropriation approp JOIN object_class_program_activity op ON approp.tas = op.tas WHERE approp.submission_id = {} GROUP BY approp.row, approp.tas, approp.grossoutlayamountbytas_cpe HAVING approp.grossoutlayamountbytas_cpe <> SUM(op.grossoutlayamountbyprogramobjectclass_cpe)",A18,"GrossOutlayAmountByTAS_CPE (file A) = sum of all GrossOutlayAmountByProgramObjectClass (file B) {When Program Activity and Object Class are not blank, this value is the sum of all Gross Outlay Amounts reported in file B, by TAS/Subaccount}",The GrossOutlayAmountByTAS_CPE amount in the appropriation file (A) does not equal the sum of the corresponding GrossOutlayAmountByProgramObjectClass_CPE values in the award financial file (B).,y,appropriations,warning
```

Comment:
For rule A18, grouping by TAS is clear since that's what the rule is checking, but I don't understand why we're grouping on row number and grossoutlayamount.  Since these are both coming from approp and there is only one line per TAS in the approp file, it seems like these would have no effect, but I'd like to understand the intent behind them.  I also still have some confusion over the sum that I'd like to discuss in slack.